### PR TITLE
jira-issue-update-build 0.1.4

### PR DESCRIPTION
**Pull request by fredyshox**
TagCheck

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  - name: git
+  - name: libgit2
+  apt_get:
+  - name: git
+  - name: libgit2-dev
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: true
+inputs:
+- APP_VERSION: null
```
Comment (imrekel) [inline steps/jira-issue-update-build/0.1.3/step.yml:39]: Could you please use lower_case ids for step inputs? UPPER_CASE should be used for outputs only.

Comment (imrekel) [reply steps/jira-issue-update-build/0.1.3/step.yml:39]: Does this mean that this step works with iOS builds only? I see `APP_VERSION` is a required input. 

If so, then this should be mentioned in the description of the step (now I'd assume that it works with any platform's builds).

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    is_sensitive: true
+    summary: User (or bot) password or API token (you can generate it [here](https://confluence.atlassian.com/cloud/api-tokens-938839638.html)).
+    title: JIRA access token
+- JIRA_ISSUE_PATTERN: ([A-Z]{1,10}-[0-9]+)
+  opts:
+    is_required: true
+    summary: A regular expression for matching issue keys in commit messages. For
+      example, ([A-Z]{1,10}-[0-9]+)
+    title: Regex pattern used to identify issue keys from commit messages
+- BITRISE_API_TOKEN: null
```
Comment (imrekel) [inline steps/jira-issue-update-build/0.1.3/step.yml:77]: What does this step use the Bitrise API for? Don't you have all build related info in the build itself? I mean these default env vars (including the build number): https://devcenter.bitrise.io/builds/available-environment-variables/

**Review by imrekel — CHANGES_REQUESTED** [review]

File: steps/jira-issue-update-build/0.1.3/step.yml (hunk 1 above)
Comment (fredyshox) [reply steps/jira-issue-update-build/0.1.3/step.yml:39]: No, it should work on both platforms. In description of this var, I only provided iOS instruction, as I don't know where it is on Android.

File: steps/jira-issue-update-build/0.1.3/step.yml (hunk 2 above)
Comment (fredyshox) [reply steps/jira-issue-update-build/0.1.3/step.yml:77]: Step is using Bitrise API, to keep JIRA issues from the commits of aborted/failed builds preceding the current one. This way if for some reason build failed, info about issues it resolved would not be lost.

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+title: Update JIRA issues with build number
+summary: Update fields of JIRA issues with build number.
+description: "Update fields of JIRA issues associated with current MR/PR, with the
```
Comment (zsolt-marta-bitrise) [inline steps/jira-issue-update-build/0.1.3/step.yml:3]: This seems very similar to the step [jira-issue-field-add-or-update](https://app.bitrise.io/integrations/steps/jira-issue-field-add-or-update). What is the different functionality that warrants a _separate_ step? Have you considered extending that step?

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  apt_get:
+  - name: git
+  - name: libgit2-dev
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: true
+inputs:
+- APP_VERSION: null
+  opts:
+    description: "If you develop an iOS project, you can extract the version number
```
Comment (zsolt-marta-bitrise) [inline steps/jira-issue-update-build/0.1.3/step.yml:41]: FYI, you can use https://app.bitrise.io/integrations/steps/get-info-on-android-app-bundle to get the version from an Android build

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      from Info.plist of the main target. It’s called `CFBundleShortVersionString`.
+      \nAlternatively, you may use a custom step that can do it for you: [bitrise-step-xcode-build-version](https://github.com/nodes-ios/bitrise-step-xcode-build-version)"
+    is_required: true
+    summary: App version (for example 1.0.0)
+    title: App version
+- JIRA_HOST: null
```
Comment (zsolt-marta-bitrise) [inline steps/jira-issue-update-build/0.1.3/step.yml:47]: Please use `lower_case` step inputs: https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md#input-naming-convention

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    title: JIRA host
+- JIRA_CUSTOM_FIELD_ID: null
+  opts:
+    description: "You can read how to find it [here](https://confluence.atlassian.com/jirakb/how-to-find-id-for-custom-field-s-744522503.html).
+      \nAnd if you don’t have a custom field yet, check out this [guide](https://confluence.atlassian.com/adminjiraserver/adding-a-custom-field-938847222.html)
+      that explains how to create it."
+    is_required: true
+    summary: Custom field id for the build number (integer)
+    title: JIRA Custom Field ID
+- JIRA_USERNAME: null
```
Comment (zsolt-marta-bitrise) [inline steps/jira-issue-update-build/0.1.3/step.yml:60]: Please only specify default values if they have definite values, and make sure they are strings. Check this please in all cases

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+- osx-10.10
+- ubuntu-16.04
+type_tags:
+- notification
+toolkit:
+  go:
+    package_name: github.com/Holdapp/bitrise-step-jira-build
+deps:
+  brew:
+  - name: git
```
Comment (zsolt-marta-bitrise) [inline steps/jira-issue-update-build/0.1.3/step.yml:30]: Git is preinstalled on the stack, so it is not required to be installed separately

File: steps/jira-issue-update-build/0.1.3/step.yml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  - name: libgit2
+  apt_get:
+  - name: git
+  - name: libgit2-dev
+is_requires_admin_user: true
```
Comment (zsolt-marta-bitrise) [inline steps/jira-issue-update-build/0.1.3/step.yml:35]: This is deprecated now

**Review by zsolt-marta-bitrise — CHANGES_REQUESTED** [review]
Hi! Please consider extending `jira-issue-field-add-or-update` instead. Added a couple of other comments in case this separate step is absolutely needed

**Comment by tadam313** [comment]
hi @fredyshox ! thanks again for the contribution, are you still planning to have a look at the change requests?

**Comment by fredyshox** [comment]
Hi sorry for delay, I was busy lately. I've spotted some issues on linux stacks and gonna try to look into this today/tomorrow

**Comment by tadam313** [comment]
Unfortunately it's not supported at the moment via step.yml. I can recommend the same resolution as @ofalvai mentioned on discuss to use script step to install the desired version inside the step

**Comment by fredyshox** [comment]
Hi, I managed to fix issues on ubuntu stacks in 0.1.4. Also applied your requested changes

**Review by zsolt-marta-bitrise — APPROVED** [review]

**Review by imrekel — APPROVED** [review]

**Pull request merged**